src/common/collection.c

```diff
@@ -1471,7 +1471,7 @@ static gchar *get_query_string(const dt_collection_properties_t property, const
                                   G_DIR_SEPARATOR_S "%%'))",
                                   escaped_text, escaped_text);
         }
-        else if ((escaped_length > 0) && (escaped_text[escaped_length-1] == '%'))
+        else if ((escaped_length > 1) && (escaped_text[escaped_length-1] == '%'))
         {
           escaped_text[escaped_length-2] = '\0';
```

Comment:
Isn't the proper fix to have a -1 here instead of -2 ? Because with your fix you just disable the case where the folder name is a simple '%'... And of course not entering the if you fix (hide) the crash.

Reply:
Not sure :man_shrugging:. Alas, I didn't manage to understand all the _raffinements_ in the query generator. There must have been some good reason for -2 I presumed, and with >1 it still displays all images both for Filmroll % and Folder %, without the (very real) segfault.

Reply:
You probably need to do some archeological work to see when & why the -2 has been introduced.

Reply:
Introduced in 86a13b6371df6c837c214699875ad38f55ab59b5 by @quovadit 
Maybe he can tell us something.

Reply:
sorry for the dumb mistake :man_facepalming:

Ctrl-Click on a folder adds |% to only show subfolders
As folders don't have the pipe as separator, I removed the last 2 characters and added /% instead
Of course the length has to be at least 2 for this to work.

So the fix by @TurboGit works, but the clean fix would be to compare the last 2 characters:

        else if ((escaped_length > 1) && (strcmp(escaped_text+escaped_length-2, "|%") == 0 ))

Reply:
@quovadit  Thanks. Will you open a PR so I can close this one?